gap/semigroups/semifp.gi

```diff
@@ -636,3 +636,160 @@ function(M)
 
   return MagmaIsomorphismByFunctionsNC(M, MF, map, inv);
 end);
+
+InstallMethod(ParseRelations,
+"for a list of free generators and a string",
+[IsDenseList, IsString],
+function(gens, inputstring)
+    local newinputstring, g, chartoel, RemoveBrackets, ParseRelation, output;
+
+    if ForAny(gens, x -> not Size(String(x)) = 1) then
+        ErrorNoReturn("multicharacter generators are not accepted");
+    fi;
+
+    if ForAny(gens, x -> String(x) in ["^", "(", ")", " ", ","]) then
+        ErrorNoReturn("the characters ^() , are not accepted as generators");
+    fi;
+
+    newinputstring := Filtered(inputstring, x -> not x = ' ');
+
+    for g in gens do
+      newinputstring := ReplacedString(newinputstring,
+                        [String(g)[1], '^'], ['(', String(g)[1], ')', '^']);
+    od;
+
+    RemoveBrackets := function(word)
+        local i, product, lbracket, rbracket, nestcount, index, chartoel;
+
+        if word = "" then
+            return ErrorNoReturn("empty product is not accepted");
+        fi;
+
+        # if the number of left brackets is different from the number of right
+        # brackets they can't possibly pair up
+        if not Size(Filtered(word, x -> x = '(')) =
+               Size(Filtered(word, x -> x = ')')) then
+            ErrorNoReturn("invalid bracket structure");
+        fi;
+
+        # if the ^ is at the end of the string there is no exponent.
+        # if the ^ is at the start of the string there is no base.
+        if word[1] = '^' or word[Size(word)] = '^' then
+            ErrorNoReturn("invalid power structure");
+        fi;
+        # checks that all ^s have an exponent.
+        for index in [1 .. Size(word)] do
+            if word[index] = '^' then
+                if not word[index + 1] in "0123456789" then
+                    ErrorNoReturn("invalid power structure");
```

Comment:
Perhaps `if power <> "to the people" then ErrorNoReturn("invalid power structure") fi;`